storage/innobase/fsp/fsp0file.cc

```diff
@@ -346,6 +346,13 @@ Datafile::read_first_page(bool read_only_mode)
 		if (!fsp_flags_is_valid(m_flags, m_space_id)) {
 			ulint cflags = fsp_flags_convert_from_101(m_flags);
 			if (cflags == ULINT_UNDEFINED) {
+				if (FSP_FLAGS_GET_PAGE_SSIZE(m_flags) == 2) { 
```

Comment:
What if the MySQL tablespace encryption (`m_flags&1<<13`) is being used, either with MySQL 5.7 or 8.0? Would the first page of the file appear completely unreadable in that case?

Can we add this check to `IMPORT TABLESPACE` as well? Could you add some Perl code to generate a dummy file that would exercise this check? You can use the test `innodb.log_corruption` for reference.